File: qualifier-week-5/graded-assignments/GrPA4 - Graded.lua
function is_prime(n)
    if n <= 1 then
        return false
    end
    for i = 2, math.sqrt(n) do
        if n % i == 0 then
            return false
        end
    end
    return true
end

function prime_sum(p, q)
    if p > q then
        return 0
    end

    local sum = 0
    for i = p, q do
        if is_prime(i) then
            sum = sum + i
        end
    end
    return sum
end

local input1 = tonumber(io.read())
local input2 = tonumber(io.read())

if not input1 or input1 <= 0 or not input2 or input2 < input1 then
    print("Error: Failure")
else
    print(prime_sum(input1, input2))
end

File: qualifier-week-5/graded-assignments/GrPA4 - Graded.go
package main

import (
	"fmt"
	"math"
)

func isPrime(num int) bool {
	if num <= 1 {
		return false
	}
	for i := 2; i <= int(math.Sqrt(float64(num))); i++ {
		if num%i == 0 {
			return false
		}
	}
	return true
}

func primeSum(p, q int) int {
	if p > q {
		return 0
	}

	var sum int
	for i := p; i <= q; i++ {
		if isPrime(i) {
			sum += i
		}
	}
	return sum
}

func main() {
	var input1, input2 int

	fmt.Scan(&input1)
	fmt.Scan(&input2)

	if input1 > input2 {
		fmt.Println("Error: Failure")
		return
	}

	println(primeSum(input1, input2))
}
